disdrodb/tests/test_config_files.py

```diff
@@ -0,0 +1,336 @@
+
+
+def is_string_list(obj: list) -> bool:
+    """Function to check that a list contains only string values
+
+    Parameters
+    ----------
+    obj : list
+        List to check
+
+    Returns
+    -------
+    bool
+        True if list contains only string values, False if not
+    """
+    if isinstance(obj, list):
+        if len(obj) == 0:
+            return True
+        else:
+            return all(isinstance(x, str) for x in obj)
+    else:
+        return False
+
+
+list_of_yaml_file_paths = list_files(CONFIG_FOLDER, "L0_data_format.yml")
+
+
+@pytest.mark.parametrize("yaml_file_path", list_of_yaml_file_paths)
+def test_L0_data_format(yaml_file_path: str):
+    """Test the format and the content of the L0_data_format.yml file.
+
+    Parameters
+    ----------
+    yaml_file_path : str
+        Path of the yaml file to test
+    """
+    data = read_yaml_file(yaml_file_path)
+
+    # check that the data is a dictionnary
+    assert is_dict(data)
+
+    # check that the second level of the dictionnary match the schema
+    for key, value in data.items():
+        assert validate_schema_pytest(value, L0_data_format_2n_level)
+
+
+# Test *_bins.yaml formatting and content
+list_of_files = ["diameter_bins.yml", "velocity_bins.yml"]
+list_of_yaml_file_paths = []
+
+for i in list_of_files:
+    list_of_yaml_file_paths.extend(list_files(CONFIG_FOLDER, i))
+
+
+@pytest.mark.parametrize("yaml_file_path", list_of_yaml_file_paths)
+def test_bins_format(yaml_file_path: str) -> None:
+    """Test the format and the content of *_bins.yml files
+
+    Parameters
+    ----------
+    yaml_file_path : str
+        Path of the yaml file to test
+    """
+    data = read_yaml_file(yaml_file_path)
+
+    # check that the data is a dictionnary
+    assert is_dict(data)
+
+    # ckeck that the keys are strings
+    list_of_fisrt_level_keys = list(data.keys())
+    assert is_string_list(list_of_fisrt_level_keys)
+
+    # check that the values are lists
+    list_of_fisrt_level_values = list(data.values())
+    assert is_list(list_of_fisrt_level_values)
+
+    # check the second level
+    for first_level_key, first_level_value in data.items():
+
+        list_of_second_level_keys = list(first_level_value.keys())
+        list_of_second_level_values = list(first_level_value.values())
+
+        # check that the keys are sorted integers
+        assert is_sorted_int_keys(list_of_second_level_keys)
+
+        # check that the values are numeric
+        if first_level_key in ["center", "width"]:
+            assert is_numeric_list(list_of_second_level_values)
+
+        if first_level_key in ["bounds"]:
+            for second_level_key, second_level_value in first_level_value.items():
+                assert is_numeric_list(second_level_value)
+
+    # check bound and width equls lenght
+    assert len(data.get("bounds")) == len(data.get("width"))
+
+    # check that the bounds distance is equal to the width (but not for the first key)

```

Comment:
Maybe we can avoid also the last key? Since the last bin is unbounded?

Reply:
ok done